src/integrate.rs

```diff
@@ -385,6 +398,39 @@ pub fn integrate<P: AsRef<Path>>(
                 kind: IntegrationErrKind::Repak(e),
             })?;
 
+        let gameplay_affecting = check_gameplay_affecting(
+            &fsd_lowercase_path_map,
+            &mut fsd_pak_reader,
+            &fsd_pak,
+            &mut buf,
+            &pak,
+        )
+        .map_err(|e| IntegrationErr {
+            mod_ctxt: Some(mod_info.clone()),
+            kind: IntegrationErrKind::Generic(e),
+        })?;
+
+        debug!(?mod_info, ?gameplay_affecting);
+
+        // HACK: if a url is like mod.io/mod-name#mod-id/mod-file-id we split right to get rid of
+        // the mod-file-id.
+        let resolution_hack = if let Some((left, _)) = mod_info.resolution.url.rsplit_once('/') {
+            ModResolution {
+                status: mod_info.resolution.status.clone(),
+                url: left.to_owned(),
+            }
+        } else {
+            mod_info.resolution.clone()
+        };
```

Comment:
~I believe this hack can go away if `ModResolution` is used as a key.~

Nevermind, I see that `ModResolution` is not actually fully resolve (as in resolves to a URL with a specific modfile ID). Maybe it should be changed to do so? I will have to look further into this.